src/Components/Posts/ParentPost.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useRef, useState, useContext } from 'react'
+import HeaderLoggedIn from '../Headers/HeaderLoggedIn'
+import PostCards from './PostCards'
+import { useLocation, useNavigate } from 'react-router-dom'
+import ContextAPI from '../../Context/ContextAPI'
+
+const ParentPost = () => {
+    const context = useContext(ContextAPI)
+    const navigate = useNavigate()
+    const location = useLocation()
+    const [postDelete, setPostDelete] = useState(false)
+    const [currentUser] = useState(location.state.isLoggedIn[0])
+    const [addNewPost, setAddNewPost] = useState(false)
+    const postTitleRef = useRef()
+    const postBodyRef = useRef()
+
+    const openOrCloseModal = () => {
+        setAddNewPost(!addNewPost)
+    }
+
+    const publishPost = (e) => {
+        e.preventDefault()
+        const newPost = {
+            title: postTitleRef.current.value,
+            body: postBodyRef.current.value,
+            id: context.postsData.length + 1,
+            userId: currentUser.id
+        }
+        context.postsData.push(newPost)
```

Comment:
Remove push instead using the set method here.

Reply:
I used Custom Hook, useFetch for getting data from endpoint and the response data is assigning to postsData. PostsData is the part of Context API and there is no useState Hook using